boards/arm/apollo4p_evb/Kconfig.board

```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+
+config BOARD_APOLLO4P_EVB
```

Comment:
note: board in the first commit is useless, add it later in the sequence (e.g. when uart can print hello world)

Reply:
Reordered the commits so that the board is added after drivers are in place.